src/crimson/common/fatal_signal.cc

```diff
@@ -42,10 +59,10 @@ void FatalSignal::install_oneshot_signal_handler()
     }
     assert(info);
     FatalSignal::signaled(signum, *info);
-    ::signal(signum, SIG_DFL);
+    reraise_fatal(signum);
   };
-  sigfillset(&sa.sa_mask);
-  sa.sa_flags = SA_SIGINFO | SA_RESTART;
+  sigemptyset(&sa.sa_mask);
+  sa.sa_flags = SA_SIGINFO | SA_RESTART | SA_NODEFER;
```

Comment:
Do you mean adding `SA_NODEFER` allows other threads raise the signal at the same time?

Reply:
Nope, it's to allow `reraise()` to signal from the signal's handler. This assures the dump will be exactly about the fault the handler was called for.